app/src/main/java/chat/rocket/android/helper/RocketChatUserAvatar.kt

```diff
@@ -1,17 +1,91 @@
 package chat.rocket.android.helper
 
+import android.content.Context
+import android.graphics.Typeface
+import android.graphics.drawable.Drawable
+import com.amulyakhare.textdrawable.TextDrawable
 import java.net.URLEncoder
 
-class RocketChatUserAvatar(val hostname: String, val username: String) {
-
-    val imageUri: String
-        /** REMARK
-         * This is often a SVG image (see Rocket.Chat:server/startup/avatar.js).
-         */
-        get() {
-            return "https://" +
-                    hostname.replace("http://", "").replace("https://", "") +
-                    "/avatar/" +
-                    URLEncoder.encode(username, "UTF-8")
+object RocketChatUserAvatar {
+
+    /**
+     * Returns the user avatar URI.
+     *
+     * REMARK: This is often a SVG image (Rocket.Chat:server/startup/avatar.js).
+     *
+     * @param hostname The server's hostname.
+     * @param username The username.
+     * @return The user avatar URI.
+     * @see getImageFormat
+     */
+    fun getUri(hostname : String, username: String): String {
+        return "https://" +
+                hostname.replace("http://", "").replace("https://", "") +
+                "/avatar/" +
+                URLEncoder.encode(username, "UTF-8")
+    }
+
+    /**
+     * Returns the user avatar image format.
+     *
+     * @param uri The user avatar image URI to get the image format.
+     * @return The user avatar image format. Possible values are: "image/jpeg", "image/png" and "image/svg+xml".
+     * @see getUri
+     */
+    fun getImageFormat(uri: String): String {
+        return OkHttpHelper.getContentType(uri)
+    }
+
+    /**
+     * Returns a drawable with username initials.
+     *
+     * @param username The username.
+     * @param context The context.
+     * @return A drawable with username initials.
+     * @see getUsernameInitials
+     */
+    fun getTextDrawable(username: String, context: Context): Drawable {
+        val round = (4 * context.resources.displayMetrics.density).toInt()
+
+        return TextDrawable.builder()
+                .beginConfig()
+                .useFont(Typeface.SANS_SERIF)
+                .endConfig()
+                .buildRoundRect(getUsernameInitials(username), getUserAvatarBackgroundColor(username), round)
+    }
+
+    /**
+     * Returns a string with the username initials. For example: username John Doe returns JD initials.
+     *
+     * @param username The username.
+     * @return A string with username initials.
+     */
+    fun getUsernameInitials(username: String): String {
```

Comment:
@luciofm Well... then I think we have a outdate documentation, because the [IOS version](https://github.com/RocketChat/Rocket.Chat.iOS/blob/develop/Rocket.Chat/Views/Avatar/AvatarView.swift) and [server version](https://github.com/filipedelimabrito/Rocket.Chat/blob/eefbd71089eac34e81d4f75f4e6c0dc07e1bd1a1/server/startup/avatar.js) only handles the separation by "." too.

Note that  we are getting the username and not the user's real name. It is right @rafaelks ?